base_layer/core/src/validation/block_validators.rs

```diff
@@ -299,20 +299,10 @@ fn check_inputs_are_utxos<B: BlockchainBackend>(block: &Block, db: &B) -> Result
     Ok(())
 }
 
-// This function checks that the inputs and outputs do not exist in the STxO set.
-fn check_not_stxos<B: BlockchainBackend>(block: &Block, db: &B) -> Result<(), ValidationError> {
-    for input in block.body.inputs() {
-        if is_stxo(db, input.hash())? {
-            // we dont want to log this as a node or wallet might retransmit a transaction
-            debug!(
-                target: LOG_TARGET,
-                "Block validation failed due to already spent input: {}", input
-            );
-            return Err(ValidationError::ContainsSTxO);
-        }
-    }
+// This function checks that the outputs do not exist in the STxO set.
+fn check_outputs_duplicate_hashes<B: BlockchainBackend>(block: &Block, db: &B) -> Result<(), ValidationError> {
     for output in block.body.outputs() {
-        if is_stxo(db, output.hash())? {
+        if is_utxo_in_mmr(db, output.hash())? {
```

Comment:
There error message in line 310 (`return Err(ValidationError::ContainsSTxO);`)  and debug message should be updated to reflect what this check is actually doing, otherwise it will be confusing to debug in future